src/server/zset_family.cc

```diff
@@ -870,61 +898,122 @@ OpResult<AddResult> OpAdd(const OpArgs& op_args, const ZParams& zparams, string_
   return aresult;
 }
 
-struct StoreArgs {
+struct UnionArgs {
   AggType agg_type = AggType::SUM;
   unsigned num_keys;
   vector<double> weights;
+  bool with_scores = false;
 };
 
-OpResult<StoreArgs> ParseStoreArgs(CmdArgList args) {
-  string_view num_str = ArgS(args, 2);
-  StoreArgs store_args;
+OpResult<void> FillAggType(UnionArgs& union_args, CmdArgList args, unsigned arg_pos) {
+  ToUpper(&args[arg_pos]);
+  string_view agg = ArgS(args, arg_pos);
+  if (agg == "SUM") {
+    union_args.agg_type = AggType::SUM;
+  } else if (agg == "MIN") {
+    union_args.agg_type = AggType::MIN;
+  } else if (agg == "MAX") {
+    union_args.agg_type = AggType::MAX;
+  } else {
+    return OpStatus::SYNTAX_ERR;
+  }
+  return OpStatus::OK;
+}
+
+OpResult<unsigned> ParseAggregate(UnionArgs& union_args, CmdArgList args, unsigned arg_pos,
+                                  bool store) {
+  if (store && arg_pos + 2 != args.size()) {
+    return OpStatus::SYNTAX_ERR;
+  }
+  if (!store && arg_pos + 3 < args.size()) {
+    return OpStatus::SYNTAX_ERR;
+  }
+  const auto& filled = FillAggType(union_args, args, arg_pos + 1);
```

Comment:
```cpp
auto res = ...

if (!res)
  return res.status();

return arg_pos + 1;
```

Reply:
Modified - part of https://github.com/dragonflydb/dragonfly/pull/420/commits/e44d27b75530be3c97e787eca68957f7b1a3f088